lib/quicly.c

```diff
@@ -2596,24 +2596,26 @@ static int mark_packets_as_lost(quicly_conn_t *conn, size_t count)
  * application timer should be set for loss detection. if no timer is required,
  * loss_time is set to INT64_MAX.
  */
-static int do_detect_loss(quicly_loss_t *ld, uint64_t largest_pn, uint32_t delay_until_lost, int64_t *loss_time)
+static int do_detect_loss(quicly_loss_t *ld, uint64_t largest_acked, uint32_t delay_until_lost, int64_t *loss_time)
 {
     quicly_conn_t *conn = (void *)((char *)ld - offsetof(quicly_conn_t, egress.loss));
     quicly_sentmap_iter_t iter;
     const quicly_sent_packet_t *sent;
-    int64_t sent_before = now - delay_until_lost;
+    int64_t time_threshold = now - delay_until_lost;
+    uint64_t packet_threshold =
+        largest_acked > QUICLY_LOSS_DEFAULT_PACKET_THRESHOLD ? largest_acked - QUICLY_LOSS_DEFAULT_PACKET_THRESHOLD : 0;
```

Comment:
I think the minimal value for `largest_acked` that we'd see is `0`, which means that `packet_threshold` would underflow and become `UINT64_MAX - 3`. Do we need to take care of that?

(and this might be another reason to switch to using `int64_t` for packet numbers :-)).

Reply:
Thanks Kazuho, I've added a comment as we discussed.